support/logger/logger.go

```diff
@@ -0,0 +1,127 @@
+package logger
+
+import (
+	"os"
+)
+
+// Logger is the base logger interface
+type Logger interface {
+	// basic messages, appends a newline (\n) after each entry
+	Info(msg string)
+
+	// basic messages, can be custom formatted, similar to fmt.Printf. User needs to add a \n if they want a newline after the log entry
+	Infof(msg string, args ...interface{})
+
+	// error messages, indicates to the logger that these messages can be handled differently (different color, special format, email alerts, etc.). The type of logger will determine what to do with these messages. The logger should NOT panic on these messages. Appends a newline (\n) after each entry.
+	Error(msg string)
+
+	// error messages, indicates to the logger that these messages can be handled differently (different color, special format, email alerts, etc.). The type of logger will determine what to do with these messages. The logger should NOT panic on these messages. User needs to add a \n if they want a newline after the log entry.
+	Errorf(msg string, args ...interface{})
+}
+
+// Fatal is a convenience method that can be used with any Logger to log a fatal error
+func Fatal(l Logger, e error) {
+	l.Info("")
+	l.Errorf("%s", e)
+	os.Exit(1)
+}
+
+/*
+Notes:
+	What to do with log lines followed by panic? Should those just go to logger.Fatal and exit?
+	What to do with utils.PriceAsFloat? If you have it return an error many of the referencing functions have to parse multiple errors.
```

Comment:
functions can have a global variable called `utilsLogger` which we set by calling `utils.SetLogger()` in `trade.go`:
```golang
func SetLogger(l logger.Logger) {
    if utilsLogger != nil {
        // protect against resetting of logger
        logger.Fatal(utilsLogger, fmt.Errorf("cannot reset the logger since it's non-nil"))
    }
    utilsLogger = l
}
```

utils.PriceAsFloat can use utilsLogger for now. This will simplify logging for this file for now.